### ISSUE TITLE
Returning from sub-orchestration fails

### ISSUE THREAD

**Issue by michaelaird** [microsoft/durabletask-mssql#145]
I'm not 100% sure this is an issue with the mssql provider but I thought I should start here.

I have a very simple (demo) function that has an orchestration that calls a sub-orchestration. When the sub-orchestration returns, I get this error.

```
[2023-01-26T16:17:53.506Z] Executed 'Functions.PrimaryFunc' (Failed, Id=6361744d-0dc1-4921-8910-87c98a876c70, Duration=277ms)

[2023-01-26T16:17:53.506Z] System.Private.CoreLib: Exception while executing function: Functions.PrimaryFunc. Microsoft.Azure.WebJobs.Host: Exception binding parameter 'context'. Microsoft.DurableTask.Sidecar.Protobuf: Value cannot be null. (Parameter 'value').

[2023-01-26T16:17:53.511Z] c91bbee0ffd04ff787fc9280ac6c01a1: Function 'PrimaryFunc (Orchestrator)' failed with an error. Reason: Microsoft.Azure.WebJobs.Host.FunctionInvocationException: Exception while executing function: Functions.PrimaryFunc

[2023-01-26T16:17:53.511Z]  ---> System.InvalidOperationException: Exception binding parameter 'context'

[2023-01-26T16:17:53.511Z]  ---> System.ArgumentNullException: Value cannot be null. (Parameter 'value')

[2023-01-26T16:17:53.511Z]    at Microsoft.DurableTask.Protobuf.SubOrchestrationInstanceCreatedEvent.set_InstanceId(String value) in /_/src/DurableTask.Sidecar.Protobuf/obj/Release/netstandard2.0/OrchestratorService.cs:line 3662

[2023-01-26T16:17:53.511Z]    at Microsoft.Azure.WebJobs.Extensions.DurableTask.ProtobufUtils.ToHistoryEventProto(HistoryEvent e) in D:\a\_work\1\s\src\WebJobs.Extensions.DurableTask\ProtobufUtils.cs:line 134

[2023-01-26T16:17:53.513Z]    at System.Linq.Enumerable.SelectListIterator`2.MoveNext()

[2023-01-26T16:17:53.513Z]    at Google.Protobuf.Collections.RepeatedField`1.AddRange(IEnumerable`1 values)

[2023-01-26T16:17:53.513Z]    at Microsoft.Azure.WebJobs.Extensions.DurableTask.OrchestrationTriggerAttributeBindingProvider.OrchestrationTriggerBinding.BindAsync(Object value, ValueBindingContext context) in D:\a\_work\1\s\src\WebJobs.Extensions.DurableTask\Bindings\OrchestrationTriggerAttributeBindingProvider.cs:line 149

[2023-01-26T16:17:53.513Z]    at Microsoft.Azure.WebJobs.Host.Indexers.FunctionIndexer.TriggerWrapper.BindAsync(Object value, ValueBindingContext context) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Indexers\FunctionIndexer.cs:line 534

[2023-01-26T16:17:53.520Z]    at Microsoft.Azure.WebJobs.Host.Triggers.TriggeredFunctionBinding`1.BindCoreAsync(ValueBindingContext context, Object value, IDictionary`2 parameters) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Triggers\TriggeredFunctionBinding.cs:line 57

[2023-01-26T16:17:53.520Z]    --- End of inner exception stack trace ---

[2023-01-26T16:17:53.520Z]    at Microsoft.Azure.WebJobs.Host.Executors.DelayedException.Throw() in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\DelayedException.cs:line 27

[2023-01-26T16:17:53.526Z]    at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ParameterHelper.PrepareParametersAsync() in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 922

[2023-01-26T16:17:53.526Z]    at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithWatchersAsync(IFunctionInstanceEx instance, ParameterHelper parameterHelper, ILogger logger, CancellationTokenSource functionCancellationTokenSource) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 492

[2023-01-26T16:17:53.526Z]    at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance, FunctionStartedMessage message, FunctionInstanceLogEntry instanceLogEntry, ParameterHelper parameterHelper, ILogger logger, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 306

[2023-01-26T16:17:53.533Z]    --- End of inner exception stack trace ---

[2023-01-26T16:17:53.533Z]    at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.ExecuteWithLoggingAsync(IFunctionInstanceEx instance, FunctionStartedMessage message, FunctionInstanceLogEntry instanceLogEntry, ParameterHelper parameterHelper, ILogger logger, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 352

[2023-01-26T16:17:53.540Z]    at Microsoft.Azure.WebJobs.Host.Executors.FunctionExecutor.TryExecuteAsync(IFunctionInstance functionInstance, CancellationToken cancellationToken) in C:\projects\azure-webjobs-sdk-rqm4t\src\Microsoft.Azure.WebJobs.Host\Executors\FunctionExecutor.cs:line 108. IsReplay: False. State: Failed. HubName: MyTaskHub. AppName: . SlotName: . ExtensionVersion: 2.9.0. SequenceNumber: 7. TaskEventId: -1
```

Please let me know if you have any ideas....

**Comment by cgillum** [comment]
It’s possible that this is a bug in the MSSQL provider. Can you provide the sample code that you used?

**Comment by michaelaird** [comment]
Here is the code

```c#
using Microsoft.Azure.Functions.Worker;
using Microsoft.Azure.Functions.Worker.Http;
using Microsoft.DurableTask;
using Microsoft.Extensions.Logging;

namespace IsolatedFunctionApp1.Untyped;

static class HelloDouble
{
    [Function(nameof(StartPrimary))]
    public static async Task<HttpResponseData> StartPrimary(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")] HttpRequestData req,
        [DurableClient] DurableClientContext durableContext,
        FunctionContext executionContext)
    {
        ILogger logger = executionContext.GetLogger(nameof(StartPrimary));

        string instanceId = await durableContext.Client.ScheduleNewOrchestrationInstanceAsync(nameof(PrimaryFunc));
        logger.LogInformation("Created new orchestration with instance ID = {instanceId}", instanceId);

        return durableContext.CreateCheckStatusResponse(req, instanceId);
    }

    [Function(nameof(PrimaryFunc))]
    public static async Task<List<string>> PrimaryFunc([OrchestrationTrigger] TaskOrchestrationContext context)
    {
        var outputs = new List<string>();

        outputs.Add(await context.CallSubOrchestratorAsync<string>(nameof(EvaluateGoal), input: new SimpleDTO("Tokyo", 1, false)));
        outputs.Add(await context.CallSubOrchestratorAsync<string>(nameof(EvaluateGoal), input: new SimpleDTO("Seattle", 2, true)));
        outputs.Add(await context.CallSubOrchestratorAsync<string>(nameof(EvaluateGoal), input: new SimpleDTO("London", 3, false)));

        return outputs;
    }

    [Function(nameof(EvaluateGoal))]
    public static async Task<string> EvaluateGoal([OrchestrationTrigger] TaskOrchestrationContext context)
    {
        SimpleDTO input = context.GetInput<SimpleDTO>();
        string result = await context.CallActivityAsync<string>(nameof(SayHello), input);
        return result;
    } 

    [Function(nameof(SayHello))]
    public static string SayHello([ActivityTrigger] SimpleDTO input, FunctionContext executionContext)
    {
        ILogger logger = executionContext.GetLogger(nameof(SayHello));
        logger.LogInformation("Saying hello to {name}", input.Name);
        return $"Hello, {input.Name}!";
    }
}

public class SimpleDTO
{
    public string Name { get; set; }
    public int Count { get; set; }
    public bool Available { get; set; }

    public SimpleDTO(string name, int count, bool available)
    {
        Name = name;
        Count = count;
        Available = available;
    }
}

```


and here is the .csproj file

```xml

<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>

    <TargetFramework>net6.0</TargetFramework>

    <AzureFunctionsVersion>v4</AzureFunctionsVersion>

    <OutputType>Exe</OutputType>

    <ImplicitUsings>enable</ImplicitUsings>

    <Nullable>enable</Nullable>

  </PropertyGroup>

    <ItemGroup>

    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.10.0" />

    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask" Version="1.0.0-rc.1" />

    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.DurableTask.SqlServer" Version="1.1.0" />

    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.13" />

    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.7.0" OutputItemType="Analyzer" />

    <PackageReference Include="Microsoft.DurableTask.Generators" Version="1.0.0-preview.1" OutputItemType="Analyzer" />

  </ItemGroup>

  <ItemGroup>

    <None Update="host.json">

      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>

    </None>

    <None Update="local.settings.json">

      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>

      <CopyToPublishDirectory>Never</CopyToPublishDirectory>

    </None>

  </ItemGroup>

  <ItemGroup>

    <Using Include="System.Threading.ExecutionContext" Alias="ExecutionContext" />

  </ItemGroup>

</Project>

```

You can see that it's pretty basic and adapted from the samples

**Comment by michaelaird** [comment]
@cgillum I tested this code using the Azure Storage provider and Azurite and it worked properly. Any ideas why it is failing for SQL?

**Comment by cgillum** [comment]
Hi @michaelaird - thanks for the sample code. I was able to reproduce this locally. I'm taking a closer look now.

**Comment by cgillum** [comment]
So this is a little awkward, but I think I found the problem: https://github.com/microsoft/durabletask-mssql/blob/22af387e8145798a130601151adb4fee5f60cc06/src/DurableTask.SqlServer/SqlUtils.cs#L130

I'm looking into how to fix this now.

**Comment by michaelaird** [comment]
Yep. That looks like the culprit! :)

**Comment by michaelaird** [comment]
Thanks for the fix! Is there a timeline for 1.1.1?

We’re looking at adopting this in a project kicking off this month.

**Comment by cgillum** [comment]
I’m hoping to do a release today or tomorrow at the latest, given that this is a high impact issue with no workaround.
